File: test/recursion.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
)

var depths = map[string]int{
	"1K":   1000,
	"10K":  10000,
	"100K": 100000,
	"1M":   1000000,
	"10M":  10000000,
}

func main() {
	for depthName, depth := range depths {
		fmt.Printf("Generating %d depth object\n", depth)

		jsonString := "{"

		for i := 1; i < depth; i++ {
			jsonString += `"next":{`
		}

		jsonString += `"next":null`

		for i := 0; i < depth; i++ {
			jsonString += "}"
		}

		fileName := fmt.Sprintf("%s_recursion.json", depthName)
		filePath := filepath.Join(filepath.Dir(os.Args[0]), fileName)

		err := os.WriteFile(filePath, []byte(jsonString), 0644)
		if err != nil {
			fmt.Printf("Error writing file: %v\n", err)
			continue
		}

		fmt.Printf("File for depth %d saved as %s\n", depth, fileName)
	}
}

File: test/recursion.py
import os

depths = {
    "1K": 1_000,
    "10K": 10_000,
    "100K": 100_000,
    "1M": 1_000_000,
    "10M": 10_000_000,
}

for depth_name, depth in depths.items():
    print(f"Generating {depth} depth object")

    json_string = "{"

    for _ in range(1, depth):
        json_string += '"next":{'

    json_string += '"next":null'

    for _ in range(depth):
        json_string += "}"

    file_name = f"{depth_name}_recursion.json"
    file_path = os.path.join(os.path.dirname(__file__), file_name)

    with open(file_path, 'w') as f:
        f.write(json_string)

    print(f"File for depth {depth} saved as {file_name}")

